src/Compilers/CSharp/Portable/Symbols/Source/SourceNamespaceSymbol.AliasesAndUsings.cs

```diff
@@ -732,7 +732,7 @@ private UsingsAndDiagnostics GetUsingsAndDiagnostics(ref UsingsAndDiagnostics? u
                                 continue;
                             }
 
-                            var flags = BinderFlags.SuppressConstraintChecks;
+                            var flags = BinderFlags.SuppressConstraintChecks | BinderFlags.InUsingDirective;
```

Comment:
>var flags = BinderFlags.SuppressConstraintChecks | BinderFlags.InUsingDirective; [](http://example.com/codeflow?start=28&length=80)

I would expect a similar change in AliasSymbol.cs

Reply:
Blocking for aliases doesn't seem necessary, as aliases don't seem to cause the same circularity problems because the resolution of the target is delayed. But I'm okay to block too, as it'll be safer.